src/main/java/org/jfrog/hudson/pipeline/common/Utils.java

```diff
@@ -75,6 +83,31 @@ public static FilePath extractRootWorkspace(StepContext context, WorkflowRun bui
         return ObjectUtils.defaultIfNull(ws, cwd);
     }
 
+    /**
+     * Walk all flow nodes and find the top most workspace action. This
+     * action would represent the root workspace for the workflow run.
+     *
+     * @param execution the execution of the workflow run.
+     * @return the root workspace from the flow node tree, or null if none exist.
+     */
+    private static FilePath extractRootWorkspaceFromFlow(FlowExecution execution) {
+        if (execution == null) {
+            return null;
+        }
+        FlowGraphWalker flowWalker = new FlowGraphWalker(execution);
+        FilePath rootPath = null;
+        for (FlowNode node : flowWalker) {
+            WorkspaceAction workspaceAction = node.getAction(WorkspaceAction.class);
+            if (workspaceAction != null) {
+                FilePath rootWorkspace = workspaceAction.getWorkspace();
+                if (rootWorkspace != null) {
+                    rootPath = rootWorkspace;
```

Comment:
We can either return here and get the most recent workspace for the build, or keep the current approach and get the true root workspace which would be the first one allocated for the workflow run. I wasn't sure which would make the most sense in this case.

For example this will work with both approaches.

```groovy
rtServer(
  id: serverId,
  url: "${env.JENKINS_ARTIFACTORY_URL}",
  username: "${env.JENKINS_ARTIFACTORY_USERNAME}",
  password: "${env.JENKINS_ARTIFACTORY_PASSWORD}"
)

dir('mavendir') {
  rtMavenRun(
    buildName: buildName,
    buildNumber: buildNumber,
    pom: 'declarative-maven-example/pom.xml',
    goals: 'clean install -U',
    deployerId: "MAVEN_DEPLOYER",
    resolverId: "MAVEN_RESOLVER"
  )
}
```

But this will only work with the current approach and would not work with the alternate of getting the most recent workspace.

```groovy
rtServer(
  id: serverId,
  url: "${env.JENKINS_ARTIFACTORY_URL}",
  username: "${env.JENKINS_ARTIFACTORY_USERNAME}",
  password: "${env.JENKINS_ARTIFACTORY_PASSWORD}"
)

ws('mavendir') {
  rtMavenRun(
    buildName: buildName,
    buildNumber: buildNumber,
    pom: 'declarative-maven-example/pom.xml',
    goals: 'clean install -U',
    deployerId: "MAVEN_DEPLOYER",
    resolverId: "MAVEN_RESOLVER"
  )
}
```